File: gradle_build/src/main/java/com/xuebinduan/gradle_build/MainActivity.java
package com.xuebinduan.gradle_build;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

/**
 * Gradle是基于任务task的，它是gradle的最小执行单元，任务之间可以依赖，任务可以做一些操作
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //BuildConfig构建后就会生成
        String applicationId = BuildConfig.APPLICATION_ID;
        boolean debug = BuildConfig.DEBUG;

    }
}

File: binder/src/main/java/com/xuebinduan/binder/MainActivity.kt
package com.xuebinduan.binder

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.xuebinduan.binder.client.Client

/**
 * RPC : Across Processes Communication
 * Binder的作用：提供一个全局服务，所谓的全局就是系统中的所有应用程序都可以访问。
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)



    }
}